content/advanced/340_appmesh_flagger/flagger_setup.md

```diff
@@ -0,0 +1,123 @@
+  * [**Required**] [AppMesh setup](/advanced/330_servicemesh_using_appmesh/appmesh_installation/install_appmesh/) completed from AppMesh Workshop.
+  * [**Required**] [increased the disk size on your Cloud9 instnace](020_prerequisites/workspace/#increase-the-disk-size-on-the-cloud9-instance) as we need to build container images for our application.
+
+
+ #### Install Flagger
+
+Add Flagger Helm repository:
+
+```bash
+helm repo add flagger https://flagger.app
+```
+
+{{< output >}}
+"flagger" has been added to your repositories
+{{< /output >}}
+
+Install Flagger's Canary CRD:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/weaveworks/flagger/master/artifacts/flagger/crd.yaml
+```
+{{< output >}}
+customresourcedefinition.apiextensions.k8s.io/canaries.flagger.app created
+customresourcedefinition.apiextensions.k8s.io/metrictemplates.flagger.app created
+customresourcedefinition.apiextensions.k8s.io/alertproviders.flagger.app created
+{{< /output >}}
+
+Deploy Flagger in the appmesh-system namespace:
+
+```bash
+helm upgrade -i flagger flagger/flagger \
+--namespace=appmesh-system \
+--set crd.create=false \
+--set meshProvider=appmesh:v1beta2 \
+--set metricsServer=http://appmesh-prometheus:9090
+```
+{{< output >}}    
+Release "flagger" does not exist. Installing it now.
+NAME: flagger
+LAST DEPLOYED: Mon Dec 14 15:23:32 2020
+NAMESPACE: appmesh-system
+STATUS: deployed
+REVISION: 1
+TEST SUITE: None
+NOTES:
+Flagger installed
+{{< /output >}}
+
+
+#### Install the App Mesh Prometheus
+
+```bash
+helm upgrade -i appmesh-prometheus eks/appmesh-prometheus \
+--wait --namespace appmesh-system
+```
+
+{{< output >}}
+Release "appmesh-prometheus" does not exist. Installing it now.
+NAME: appmesh-prometheus
+LAST DEPLOYED: Sat Mar 13 20:59:29 2021
+NAMESPACE: appmesh-system
+STATUS: deployed
+REVISION: 1
+TEST SUITE: None
+NOTES:
+AWS App Mesh Prometheus installed!
+{{< /output >}}
+
+#### Enable horizontal pod auto-scaling
+
+Install the Horizontal Pod Autoscaler (HPA) metrics provider:
+```bash
+kubectl apply -f https://github.com/kubernetes-sigs/metrics-server/releases/download/v0.4.1/components.yaml
+```
+
+{{< output >}}
+serviceaccount/metrics-server created
+clusterrole.rbac.authorization.k8s.io/system:aggregated-metrics-reader created
+clusterrole.rbac.authorization.k8s.io/system:metrics-server created
+rolebinding.rbac.authorization.k8s.io/metrics-server-auth-reader created
+clusterrolebinding.rbac.authorization.k8s.io/metrics-server:system:auth-delegator created
+clusterrolebinding.rbac.authorization.k8s.io/system:metrics-server created
+service/metrics-server created
+deployment.apps/metrics-server created
+apiservice.apiregistration.k8s.io/v1beta1.metrics.k8s.io created
+{{< /output >}}
+
+After a minute, the metrics API should report CPU and memory usage for pods. You can verify the metrics API:
```

Comment:
kubectl get deployments might not be the appropriate method to show the status of metrics API, we might want to check the status of the API along with deployment.

Also possible rephrase, "You can verify status of the metrics API:"

kubectl get apiservice v1beta1.metrics.k8s.io -o json | jq '.status'

{
  "conditions": [
    {
      "lastTransitionTime": "2020-11-10T06:39:13Z",
      "message": "all checks passed",
      "reason": "Passed",
      "status": "True",
      "type": "Available"
    }
  ]
}

Reply:
Done